js/src/admin/utils/ExtensionData.js

```diff
@@ -1,34 +1,159 @@
 import ItemList from '../../common/utils/ItemList';
-import listItems from '../../common/helpers/listItems';
 
 export default class ExtensionData {
-  constructor(extension) {
-    this.extension = extension;
-    app.extensionData[extension] = app.extensionData[extension] || {};
+  constructor() {
+    this.data = {};
+    this.currentExtension = null;
   }
 
-  registerData(type, content, permissionType = null, priority = 0) {
-    if (content instanceof ItemList) {
-      content = listItems(content.toArray());
-    }
+  /**
+   * This function simply takes the extension id
+   *
+   * @example
+   * app.extensionData.load('flarum-tags')
+   *
+   * flarum/flags -> flarum-flags | acme/extension -> acme-extension
+   *
+   * @param extension
+   */
+  load(extension) {
+    this.currentExtension = extension;
+    this.data[extension] = this.data[extension] || {};
 
-    app.extensionData[this.extension][type] = app.extensionData[this.extension][type] || {};
+    return this;
+  }
 
-    if (type === 'permissions') {
-      if (!app.extensionData[this.extension][type][permissionType]) {
-        app.extensionData[this.extension][type][permissionType] = new ItemList();
-      }
-      app.extensionData[this.extension][type][permissionType].add(content.permission, content);
-    } else {
-      app.extensionData[this.extension][type] = content;
+  /**
+   * This function registers your settings with Flarum
+   *
+   * @example - settings
+   *
+   * .registerSettings('settings', {
+   *     'flarum-flags.guidelines_url': {
+   *         type: 'text', // This will be inputted into the input tag for the setting (text/number/etc)
+   *         label: app.translator.trans('flarum-flags.admin.settings.guidelines_url_label')
+   *     }
+   * })
+   *
+   *
+   * @param type
+   * @param content
+   * @param permissionType
+   * @param priority
+   * @returns {ExtensionData}
+   */
+  registerSettings(content, permissionType = null, priority = 0) {
```

Comment:
If we're registering settings by key, I think it might be best to do it in individual calls of `registerSetting`. One advantage of keeping it as `registerSettings` is that we could pass in an **ordered** array of settings instead of an unordered object.

Reply:
To clarify, I think that settings registration would make a bit more sense as an array, since that way extension settings can be added in a certain order. Alternatively, settings could be added individually (instead of as one big object), which would also allow ordered registration of settings